ansible/roles/test/templates/vnet_config.j2

```diff
@@ -15,53 +17,69 @@ intf_list:
     ip: '{{ index }}.{{ index }}.10.1/24'
 
 vnet_id_list:
-  - Vnet1
-  - Vnet2
-  - Vnet3
-  - Vnet4
+{% for vnet in range (1, 1 + num_vnet) %}
+  - Vnet{{ vnet }}
+{% endfor %}
+
+vnet_id_v4_list:
+{% for vnet_v4 in range (1, vnet_v6_base) %}
+  - Vnet{{ vnet_v4 }}
+{% endfor %}
+
+vnet_id_v6_list:
+{% for vnet_v6 in range (vnet_v6_base, 1 + num_vnet) %}
+  - Vnet{{ vnet_v6 }}
+{% endfor %}
+
 
 vnet_peer_list:
   - Vnet3: Vnet4
   - Vnet4: Vnet3
+  - Vnet7: Vnet8
+  - Vnet8: Vnet7
 
 vnet_intf_list:
-  - ifname: Vlan2001
-    vnet: Vnet1
-  - ifname: Vlan2002
-    vnet: Vnet2
-  - ifname: Vlan3001
-    vnet: Vnet3
-  - ifname: Vlan3002
-    vnet: Vnet4
+{% for vlan in range (vlan_id_base, vlan_id_base + num_vnet) %}
+  - ifname: Vlan{{ vlan }}
+    vnet: Vnet{{ loop.index }}
+{% endfor %}
 
 vnet_nbr_list:
-  - ifname: Vlan2001
-    ip: '1.1.10.101'
-  - ifname: Vlan2001
-    ip: '1.1.10.102'
-  - ifname: Vlan2002
-    ip: '2.2.10.101'
-  - ifname: Vlan2002
-    ip: '2.2.10.102'
-  - ifname: Vlan3001
-    ip: '3.3.10.101'
-  - ifname: Vlan3002
-    ip: '4.4.10.101'
+{% for vlan in range (vlan_id_base, vlan_id_base + num_vnet) %}
+  - ifname: Vlan{{ vlan }}
+    ip: '{{ loop.index }}.{{ loop.index }}.10.101'
+{% if (vlan == vlan_id_base) or (vlan == vlan_id_base + 1) 
```

Comment:
This check looks to be specific. For a better clarity, we can set any extras as additional entries after the loop.